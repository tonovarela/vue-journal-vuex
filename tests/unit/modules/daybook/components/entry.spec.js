import {shallowMount} from  '@vue/test-utils';
import Entry from '@/modules/daybook/components/Entry.vue';
import { journalState } from '../../../mock-data/test-journal-state';

describe('Pruebas en el Entry component',()=>{
  const mockRouter = {
      push:jest.fn()
  }
  const wrapper = shallowMount(Entry , {
    props:{
        entry:journalState.entries[0]
    },
    global: {
        mocks:{
            $router: mockRouter
        }
    }
  })
    test('Debe de hacer match con el snapshot',()=>{
           expect(wrapper.html()).toMatchSnapshot();
    })
    test('Debe de redireccionar al hacer click en el entry-container',()=>{
          const entryContainer = wrapper.find('.entry-container')
          entryContainer.trigger('click');
          expect(mockRouter.push).toHaveBeenCalledWith({
              name:'entry',
              params: {
                  id:journalState.entries[0].id
              }
          })
    })

    test('Pruebas en las propiedades computadas',()=>{

        expect(wrapper.vm.day).toBe(25);
        expect(wrapper.vm.month).toBe('Marzo');
        expect(wrapper.vm.yearDay).toBe(' 2022 Viernes ');

    })

});