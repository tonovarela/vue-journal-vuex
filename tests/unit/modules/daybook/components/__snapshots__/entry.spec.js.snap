// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Entry component Debe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">25</span><span class="mx-1 fs-5">Marzo</span><span class="mx-2 fw-light"> 2022 Viernes </span></div>
  <div class="entry-description">Hola desde MocKData</div>
</div>
`;
